### PR TITLE
tests: cover all possible scenarios in upgrading tests

### PR DESCRIPTION
**Pull request by lichunzhu**
### What problem does this PR solve? 
Try to fix part of https://github.com/pingcap/dm/issues/1667

### What is changed and how it works?
Add uncovered cases in https://github.com/pingcap/dm/issues/1667.

### Check List 

Tests 

 - Unit test
 - Integration test

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

**Comment by lance6716** [comment]
mariaDB is partly added in https://github.com/pingcap/dm/pull/1661, could replace it

File: tests/tiup/lib.sh [hunk 1]
```diff
@@ -115,6 +149,23 @@ function exec_incremental_stage2() {
...
+	# pessimistic shard ddls
+	exec_sql mysql1 3306 "ALTER TABLE $DB5.$TBL2 ADD COLUMN c3 INT;"
+	exec_sql mariadb2 3306 "ALTER TABLE $DB6.$TBL2 ADD COLUMN c3 INT;"
+
+	# prepare pessimistic incremental data
+	exec_sql mysql1 3306 "INSERT INTO $DB5.$TBL1 (c1, c2, c3) VALUES (203, '203', 203);"
+	exec_sql mysql1 3306 "INSERT INTO $DB5.$TBL2 (c1, c2, c3) VALUES (204, '204', 204);"
+	exec_sql mariadb2 3306 "INSERT INTO $DB6.$TBL2 (c1, c2, c3) VALUES (213, '213', 213);"
+	exec_sql mariadb2
+  3306 "INSERT INTO $DB6.$TBL3 (c1, c2, c3) VALUES (214, '214', 214);"
```
Comment (lichunzhu) [inline tests/tiup/lib.sh:168]:
```suggestion
	exec_sql mariadb2 3306 "INSERT INTO $DB6.$TBL3 (c1, c2, c3) VALUES (214, '214', 214);"
```

File: tests/tiup/upgrade-from-v2.sh [hunk 2]
```diff
@@ -53,6 +58,12 @@ function upgrade_to_current_v2() {
 }
 
 function migrate_in_v2 {
+	run_dmctl_with_retry $CUR_VER "query-status" "Running" 2 "Paused" 1
```
Comment (lance6716) [inline tests/tiup/upgrade-from-v2.sh:61]: I also want to check `show-ddl-locks` and `query-status` here 😂

**Review by lance6716 — COMMENTED** [review]
> [ ] start-task could continue from checkpoint at sync stage

It's better we `stop-task` before upgrading, and check `start-task` after upgrading works fine.

> [ ] relay enabled

I think this is not covered, so unmarked it in original issue

**Comment by lichunzhu** [comment]
> It's better we `stop-task` before upgrading, and check `start-task` after upgrading works fine.
Could you please give me a reason? I think `pause-task` can test both `pause-task will not resume automaticlly, and resume-task could work` and `recover from checkpoint in sync stage`
> I think this is not covered, so unmarked it in original issue
One source has relay enabled.

**Comment by lance6716** [comment]
>  I think pause-task can test both pause-task will not resume automaticlly, and resume-task could work and recover from checkpoint in sync stage

good idea

> One source has relay enabled 

not found `start-relay`

**Review by lance6716 — APPROVED** [review]

**Pull request merged**